### PR TITLE
Fix/improved navigation toggler

### PR DESCRIPTION
**Pull request by apfelfabrik**
On small screens, users tend to actually collapse the navigation tree. Annoyingly, a collapsed menu is open again when navigating to another page. This pull request refactors the bloated toggler and adds sessionStorage to store expansion state. Tested in current Chrome, a rather current Firefox (~13) and IE8.

### CODE REVIEW

**Comment by apfelfabrik** [comment]
I knew that @travisbot would hurt my feelings sooner or later. Please consider that the build was failing before, and I will not be able to fix it in the scope of this pull request.

File: public/themes/bsp [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../theme-generator/themes/bsp
```
Comment (meineerde) [inline public/themes/bsp]: This file should not be here.

File: public/themes/dtag [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../theme-generator/themes/dtag
```
Comment (meineerde) [inline public/themes/dtag]: This one neither.

File: public/javascripts/application.js [hunk 3]
```diff
@@ -722,25 +722,32 @@ $(window).bind('resizeEnd', function() {
...
-        jQuery('#main-menu #toggle-project-menu a.navigation-toggler').click(function(){
-          if ($('#main-menu #toggle-project-menu').hasClass('show')) {
-            // Show project navigation
-            $('#main-menu').removeClass('hidden')
-            $('#menu-sidebar').removeClass('hidden');
-            $('#main-menu #toggle-project-menu').removeClass('show');
-            $('#main-menu #toggle-project-menu').removeAttr("style");
-            $('#content').removeClass('hidden-navigation');
+        // users of some old IEs are out of luck ATM. An userData
+        // implementation could be provided though, that would be great!
```
Comment (meineerde) [inline public/javascripts/application.js:726]: You could use cookies as a fall-back.

Comment (apfelfabrik) [inline public/javascripts/application.js:726]: I could. But I don't like a trivial thing like this to pollute every forthcoming HTTP request. Besides, all supported browsers have sessionStorage. I would rather add userData for IE7 and lower, and test again in Opera, which should support sessionStorage, too. Anyway, for those browsers the behaviour shouldn't change. Read: I hate cookies.

Users on Konqueror will run into graver problems before this one, imho.

Comment (meineerde) [inline public/javascripts/application.js:726]: What I meant was to use cookies for browsers where you don't have `sessionStorage` (or similar technologies). I let you decide if it it worth it at all (as I don't now which browsers actually provide this) :)

**Comment by apfelfabrik** [comment]
Thanks for the feedback, @meineerde. Those two files somehow slipped through. It is fixed now.

My suggestion: With the next refactoring of some feature requiring client data storage, I will introduce some backend-agnistic storage, and it will have a cookie fallback. For this minor feature alone, that's overkill.

**Comment by apfelfabrik** [comment]
Rebased once more.

**Comment by apfelfabrik** [comment]
Guys?

**Comment by schmidt** [comment]
Thanks @meineerde and @apfelfabrik for contributing to this feature. The proposed changes have been merged (with minor changes) to the openproject master.

Gregor

**Pull request closed without merging**